sources/fr/chireads.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import requests
+from lncrawl.core.crawler import Crawler
+
+logger = logging.getLogger(__name__)
+
+
+class Chireads(Crawler):
+    base_url = ["https://chireads.com/"]
+    has_manga = False
+    machine_translation = False
+
+    def search_novel(self, query):
+        query = query.lower().replace(" ", "+")
+
+        # NOTE: Using self.get_soup() here throw an error, I don't know why.
+        response = requests.get("https://chireads.com/search?x=0&y=0&name=" + query)
+        soup = self.make_soup(response)
+
+        result = []
+        content = soup.find("div", {"id": "content"})
+
+        for novel in content.find_all("li"):
+            content = novel.find("a")
+            result.append({
+                "title": content.get("title"),
+                "url": self.absolute_url(content.get("href")),
+            })
+
+        return result
+
+    def read_novel_info(self):
+        soup = self.get_soup(self.novel_url)
+        content = soup.find("div", {"class": "conteiner"}).find_all(
+            "div", {"class": "wid"}
+        )
+
+        metadata = content[0]
+        
+        self.novel_cover = self.absolute_url(metadata.find("img").get("src"))
+        
+        self.novel_title = metadata.find("h3", {
```

Comment:
self.cleaner is not necessary. novel title, author, titles in chapters and volumes lists gets formatted later automatically